server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/ProgramNode.java

```diff
@@ -145,15 +148,28 @@ private void addPredefinedVariables() {
       addVariableDefinition(new MnemonicNameNode(location, PREDEFINED, predefinedVariableName));
   }
   /**
-   * Add a paragraph or section definition name node in the program context.
+   * Add a paragraph definition name node in the program context.
    *
-   * @param node - the paragraph or section definition node
+   * @param node - the section definition node
    * @return syntax error if the code block duplicates
    */
-  public Optional<SyntaxError> registerNameNode(CodeBlockNameNode node) {
-    blockReference.putIfAbsent(node.getName(), new CodeBlockReference());
-    CodeBlockReference references = blockReference.get(node.getName());
-    // TODO: add uniqueness check for paragraphs and section definition
+  public Optional<SyntaxError> registerParagraphNameNode(ParagraphNameNode node) {
+    paragraphMap.putIfAbsent(node.getName(), new CodeBlockReference());
+    CodeBlockReference references = paragraphMap.get(node.getName());
+    // TODO: add uniqueness check for paragraphs definition
+    references.addDefinition(node.locality.toLocation());
+    return Optional.empty();
+  }
+  /**
+   * Add a section definition name node in the program context.
+   *
+   * @param node - the section definition node
+   * @return syntax error if the code block duplicates
+   */
+  public Optional<SyntaxError> registerSectionNameNode(SectionNameNode node) {
+    sectionMap.putIfAbsent(node.getName(), new CodeBlockReference());
+    CodeBlockReference references = sectionMap.get(node.getName());
+    // TODO: add uniqueness check for section definition
     references.addDefinition(node.locality.toLocation());
     return Optional.empty();
```

Comment:
The only difference in logic is the map, so you can just extract a private method and pass the required map there.